### ISSUE TITLE
TrajOpt Ifopt unit test fails on Ubuntu 22.04

### ISSUE THREAD

**Issue by Levi-Armstrong** [tesseract-robotics/trajopt#364]
@rjoomen Would you be able to take a look at why the unit test is failing on 22.04?

**Comment by rjoomen** [comment]
Sure, I have time to look into it this Friday

**Comment by rjoomen** [comment]
I'm not sure, all I could find so far was that the first solve step already produces differences in the new_approx values for two of the orientational components of the cartesian position constraint. After the first two steps in the image below, the solver starts failing on 22.04.
![image](https://github.com/tesseract-robotics/trajopt/assets/1153434/2742321d-80fc-4014-9714-4794ca9ab107)

**Comment by rjoomen** [comment]
But maybe interesting is that changing the orientation to (numerical_ik_unit.cpp:97): 
```
target_pose.linear() = Eigen::Quaterniond(0.000001, 0.000001, 1, 0.000001).toRotationMatrix();
```
makes the solver also fail on 20.04.

**Comment by Levi-Armstrong** [comment]
I did a little digging and you can see that both fail with unfeasible error, but Jammy prints a console bridge error which means somethings is different. In the focal version it increases the trust region while the jammy version decreases.

Focal:
![focal](https://github.com/tesseract-robotics/trajopt/assets/9803128/5ae3d3a8-6412-48e5-9fe5-3651aac6534d)

Jammy:
![Jammy](https://github.com/tesseract-robotics/trajopt/assets/9803128/0be7a0d4-d2f2-4027-ab94-fba53059d3ca)

**Comment by Levi-Armstrong** [comment]
Addressed by #366
